src/libsystemd/sd-id128/sd-id128.c

```diff
@@ -253,17 +253,21 @@ _public_ int sd_id128_get_invocation(sd_id128_t *ret) {
         assert_return(ret, -EINVAL);
 
         if (sd_id128_is_null(saved_invocation_id)) {
-                /* We first check the environment. The environment variable is primarily relevant for user
-                 * services, and sufficiently safe as long as no privilege boundary is involved. */
-                r = get_invocation_from_environment(&saved_invocation_id);
-                if (r < 0 && r != -ENXIO)
-                        return r;
```

Comment:
Please do not change the order of the two `get_invocation_from_xxx()`.
I guess the issue can be fixed by fixing the return value handling:
```
r = get_invocation_from_environment();
if (r != -ENXIO)
        return r;
```